src/main/java/fi/aalto/cs/apluscourses/model/Module.java

```diff
@@ -0,0 +1,103 @@
+package fi.aalto.cs.apluscourses.model;
+
+import fi.aalto.cs.apluscourses.utils.Event;
+import fi.aalto.cs.apluscourses.utils.StateMonitor;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONObject;
+
+public class Module {
+
+  public static final int ERROR = StateMonitor.ERROR;
+  public static final int NOT_INSTALLED = StateMonitor.INITIAL;
+  public static final int FETCHING = NOT_INSTALLED + 1;
+  public static final int FETCHED = FETCHING + 1;
+  public static final int LOADING = FETCHED + 1;
+  public static final int LOADED = LOADING + 1;
+  public static final int WAITING_FOR_DEPS = LOADED + 1;
+  public static final int INSTALLED = WAITING_FOR_DEPS + 1;
```

Comment:
would it be of any help?
```java
  public enum States {
    ERROR(StateMonitor.ERROR),
    NOT_INSTALLED(StateMonitor.INITIAL),
    FETCHING(NOT_INSTALLED.code + 1);
    ...

    public final int code;

    private States(int code) {
      this.code = code;
    }
  }
```

Reply:
Probably, yes. I have to admit that when writing this I just didn't remember how Java enums work, and I was too lazy to start studying them. I'll check if enum can be used here, and change it, if it works well in this case.

Reply:
Tried to change to enum, but conversions between int and enum values got overly complicated, so i'd suggest we keep it old-school.